### PR TITLE
feat: [COR-1479] attempt to make websocket not do disconnection hands…

### PR DESCRIPTION
**Pull request by hoangvubui**
…hake when dispose

### CODE REVIEW

File: Bayeux@CodeTitans/BayeuxConnection.cs [hunk 1]
```diff
@@ -915,10 +915,10 @@ protected void SendRequest(BayeuxRequest message, bool asynchronous)
...
         /// </summary>
         public void Cancel()
-        {
-            if(DefaultConnectionType != BayeuxConnectionTypes.WebSocket)
-            {
-                if (_httpConnection.IsActive)
+        {
+            if(DefaultConnectionType != BayeuxConnectionTypes.WebSocket)
+            {
+                if (_httpConnection.IsActive)
```
Comment (hoangvubui) [inline Bayeux@CodeTitans/BayeuxConnection.cs:921]: possible tab conversions here, my local version has spaces afaic

File: websocket-sharp/DestroyOnDisposeWebSocket.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        /// <summary>
+        /// Closes the WebSocket connection, and releases all associated resources, without attempting
+        /// the disconnection handshake
+        /// </summary>
+        /// <remarks>
+        /// This method closes the connection with <see cref="CloseStatusCode.Away"/>.
+        /// </remarks>
+        void IDisposable.Dispose ()
+        {
+            close (new CloseEventArgs (CloseStatusCode.Away), false, true);
```
Comment (karlharmer-pm) [inline websocket-sharp/DestroyOnDisposeWebSocket.cs]: why not set the wait flag to false too here? Is there any benefit in potentially blocking the thread (granted it is not on the main thread) when we might not care about the result

**Comment by karlharmer-pm** [comment]
Have you got this bayeux/websocket lib in to an application (such as cashman) yet?

File: websocket-sharp/DestroyOnDisposeWebSocket.cs (hunk 2 above)
Comment (hoangvubui) [reply websocket-sharp/DestroyOnDisposeWebSocket.cs]: Nope, there isn't. The bool there doesn't take any effect

        send = send && _readyState == WebSocketState.Open;
        wait = wait && send;

So i just ignored it. I could change it to false.

**Review by florintoma — APPROVED** [review]

**Comment by hoangvubui** [comment]
Tested the change with hov and cashman - connectivity worked properly, although I couldn't reproduce the problem prior to this patch. I talked to Tiago and Dragos, and we agreed to put the change in and continue monitoring the crash.

**Pull request merged**